src/erldns_zone_cache.erl

```diff
@@ -249,7 +346,73 @@ delete_zone_records(Name) ->
   erldns_storage:select_delete(zone_records, [{{{erldns:normalize_name(Name), '_'}, '_'},[],[true]}]),
   erldns_storage:select_delete(zone_records_typed, [{{{erldns:normalize_name(Name), '_', '_'}, '_'},[],[true]}]).
 
-
+%% @doc Remove zone RRSet
+-spec delete_zone_rrset(binary(), binary(), binary(), integer()) -> any().
+delete_zone_rrset(Name, Digest, RRFqdn, Type) ->
+  delete_zone_rrset(Name, Digest, RRFqdn, Type, 0).
+
+%% @doc Remove zone RRSet
+-spec delete_zone_rrset(binary(), binary(), binary(), integer(), integer()) -> any().
+delete_zone_rrset(ZoneName, Digest, RRFqdn, Type, Counter) ->
+  case find_zone_in_cache(erldns:normalize_name(ZoneName)) of
+    {ok, Zone} -> 
+	  Zone,
```

Comment:
You shouldn't need this as the `Zone` is matched above and thus available in the block.

Reply:
the zone is matched in put_zone_rrset/4 and we can pass it to delete_zone_rrset/4 within that function, but this is not the case when we call delete_zone_rrset/5 on RRset's delete operation from zoneclient, hence the explicit call within the function.